src/Aspire.Hosting.Maui/MauiWindowsExtensions.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Lifecycle;
+using Aspire.Hosting.Maui;
+using Aspire.Hosting.Maui.Annotations;
+using Aspire.Hosting.Maui.Lifecycle;
+using Aspire.Hosting.Maui.Utilities;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Windows platform resources to MAUI projects.
+/// </summary>
+public static class MauiWindowsExtensions
+{
+    /// <summary>
+    /// Adds a Windows device resource to run the MAUI application on the Windows platform.
+    /// </summary>
+    /// <param name="builder">The MAUI project resource builder.</param>
+    /// <param name="name">The name of the Windows device resource. If not provided, defaults to "{projectName}-windows".</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    /// <remarks>
+    /// This method creates a new Windows platform resource that will run the MAUI application
+    /// targeting the Windows platform using <c>dotnet run</c>. The resource does not auto-start 
+    /// and must be explicitly started from the dashboard by clicking the start button.
+    /// <para>
+    /// Multiple Windows device resources can be added to the same MAUI project if needed, each with
+    /// a unique name.
+    /// </para>
+    /// </remarks>
+    /// <example>
+    /// Add a Windows device to a MAUI project:
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    /// 
+    /// var maui = builder.AddMauiProject("mauiapp", "../MyMauiApp/MyMauiApp.csproj");
+    /// var windowsDevice = maui.AddWindowsDevice();
+    /// 
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    public static IResourceBuilder<MauiWindowsPlatformResource> AddWindowsDevice(
+        this IResourceBuilder<MauiProjectResource> builder,
+        [ResourceName] string? name = null)
```

Comment:
Create an overload of this method that takes non name instead of introducing a nullable parameter here. You'll thank me later ;)

Reply:
Ooooh I see... Yeah thats better!